streaming/index.js

```diff
@@ -182,14 +181,69 @@ const CHANNEL_NAMES = [
 ];
 
 const startServer = async () => {
+  const pgPool = new pg.Pool(pgConfigFromEnv(process.env));
+  const server = http.createServer();
+  const wss = new WebSocket.Server({ noServer: true });
+
+  // Set the X-Request-Id header on WebSockets:
+  wss.on("headers", function onHeaders(headers, req) {
+    headers.push(`X-Request-Id: ${req.id}`);
+  });
+
   const app = express();
 
   app.set('trust proxy', process.env.TRUSTED_PROXY_IP ? process.env.TRUSTED_PROXY_IP.split(/(?:\s*,\s*|\s+)/) : 'loopback,uniquelocal');
 
-  const pgPool = new pg.Pool(pgConfigFromEnv(process.env));
-  const server = http.createServer(app);
   app.use(cors());
 
+  // Handle eventsource & other http requests:
+  server.on('request', app);
+
+  // Handle upgrade requests:
+  server.on('upgrade', function upgrade(request, socket, head) {
+    /** @param {Error} err */
+    const onSocketError = (err) => {
+      log.error(`Error with websocket upgrade: ${err}`);
+    };
+
+    socket.on('error', onSocketError);
+
+    // Authenticate:
+    accountFromRequest(request).then(( /* resolvedAccount */) => {
+      wss.handleUpgrade(request, socket, head, function done(ws) {
+        // Remove the error handler:
+        socket.removeListener('error', onSocketError);
+
+        // Start the connection:
+        wss.emit('connection', ws, request);
+      });
+    }).catch((err) => {
```

Comment:
Shouldn't the error handling only be covering the part before the `handleUpgrade`? i.e. just the `accountFromRequest` part.

Also, this is pretty minor, but I'd have the catch function be a separate helper function.

Reply:
Yeah, I might switch this to async/await actually

Reply:
The helper function is less about reuse than seeing the flow at a glance. I think something like `ws`'s `abortHandshake` makes sense.

Reply:
I think in future refactoring this would be a good idea, but for now, I'd be inclined to say that this would just add to the chaos of the `index.js` file that already needs significant refactoring to split it up down the road.

perhaps a compromise would be adding a "TODO" comment for future refactorer?

Reply:
I've skipped the abortHandshake change, as it'll make landing #27828 much more difficult.